django/contrib/postgres/fields/ranges.py

```diff
@@ -204,7 +208,16 @@ def db_type(self, connection):
         return "daterange"
 
 
-RangeField.register_lookup(lookups.DataContains)
+class RangeContains(lookups.DataContains):
+    def as_postgresql(self, compiler, connection):
+        sql, params = super().as_postgresql(compiler, connection)
+        if params and not isinstance(params[0], Range):
+            cast_type = self.lhs.field.base_field.cast_db_type(connection)
+            sql = sql.replace("%s", f"%s::{cast_type}")
+        return sql, params
```

Comment:
This is probably unnecessary after e0ae1363ec2aa71945be26f869cafd4181ccbc95.

Reply:
Ok will check that, is there any chance we can get #15831 reviewed and merged then rebasing stuff will get easier :D

Reply:
It's next on my list after #15922, reviewing changes in the ORM internals is ... non-trivial :smiling_face_with_tear:

Reply:
Just checked, it still seems to be required. Seems we do not call `get_placeholder` as required. Will investigate :)

Reply:
Did you investigate?

Reply:
No, didn't get around to it yet. Any chance of getting some help with that :)